axelor-human-resource/src/main/java/com/axelor/apps/hr/service/project/ProjectPlanningTimeServiceImpl.java

```diff
@@ -555,4 +559,100 @@ protected ProjectConfig getProjectConfig(ProjectPlanningTime projectPlanningTime
+  public void updateProjectPlannings(Sprint sprint, List<ProjectTask> projectTaskList) {
+
+    if (!isValidSprint(sprint) || CollectionUtils.isEmpty(projectTaskList)) {
+      return;
+    }
+
+    for (ProjectTask projectTask : projectTaskList) {
+
+      if (!isValidProjectTask(projectTask)) {
+        return;
+      }
+
+      List<ProjectPlanningTime> planningTimeList = Collections.emptyList();
+
+      ProjectTask projectTaskDb =
+          Optional.ofNullable(projectTask.getId()).map(projectTaskRepo::find).orElse(null);
+
+      if (projectTaskDb != null && projectTaskDb.getSprint() != null) {
+        Employee employeeDb =
+            Optional.of(projectTaskDb.getAssignedTo()).map(User::getEmployee).orElse(null);
+        planningTimeList = fetchPlanningList(projectTaskDb.getSprint(), employeeDb, projectTask);
+      }
+
+      if (CollectionUtils.isNotEmpty(planningTimeList)) {
+        planningTimeList.forEach(
+            planningTime -> {
+              updatePlanningTime(sprint, projectTask, planningTime);
+              planningTimeRepo.save(planningTime);
+            });
+      } else if (CollectionUtils.isEmpty(
+          fetchPlanningList(sprint, projectTask.getAssignedTo().getEmployee(), projectTask))) {
+        planningTimeRepo.save(createPlanningTime(sprint, projectTask));
+      }
+    }
+  }
+
+  protected List<ProjectPlanningTime> fetchPlanningList(
+      Sprint sprint, Employee employee, ProjectTask projectTask) {
+
+    return Query.of(ProjectPlanningTime.class)
+        .filter(
+            "self.projectTask = :projectTask and self.employee = :employee and DATE(self.startDateTime) = :startDate and DATE(self.endDateTime) = :endDate")
+        .bind("projectTask", projectTask)
+        .bind("employee", employee)
+        .bind("startDate", sprint.getFromDate())
+        .bind("endDate", sprint.getToDate())
+        .fetch();
+  }
+
+  protected ProjectPlanningTime updatePlanningTime(
+      Sprint sprint, ProjectTask projectTask, ProjectPlanningTime planningTime) {
+
+    Employee employee = projectTask.getAssignedTo().getEmployee();
+
+    planningTime.setStartDateTime(sprint.getFromDate().atStartOfDay());
+    planningTime.setEndDateTime(sprint.getToDate().atStartOfDay());
+    planningTime.setEmployee(employee);
+    planningTime.setProduct(employee.getProduct());
+
+    return planningTime;
+  }
+
+  protected ProjectPlanningTime createPlanningTime(Sprint sprint, ProjectTask projectTask) {
+
+    ProjectPlanningTime planningTime = new ProjectPlanningTime();
+
+    Employee employee = projectTask.getAssignedTo().getEmployee();
+
+    planningTime.setProjectTask(projectTaskRepo.find(projectTask.getId()));
+    planningTime.setProject(projectTask.getProject());
+    planningTime.setEmployee(employee);
+    planningTime.setStartDateTime(sprint.getFromDate().atStartOfDay());
+    planningTime.setEndDateTime(sprint.getToDate().atStartOfDay());
+    planningTime.setProduct(employee.getProduct());
```

Comment:
You need to set plannedTime by the budgetedTime then throw the same process as in ProjectPlanningTimeController.computePlannedTime

You will need to add all this process in the ProjectPlanningTimeServiceImpl